src/protocols/ssh/click.c

```diff
@@ -0,0 +1,457 @@
+        
+        /* get border */
+        guac_click* selection = guac_click_get_word_border(click);
+
+        /* Copy to clipboard */
+        guac_click_select_word(selection);
+
+        /* Draw selection */
+        guac_click_draw_select(selection);
+
+        return;
+
+    }
+
+    /* Marks */
+
+    else if ((sentinel > 32 && sentinel < 48) || (sentinel > 57 && sentinel < 65) || (sentinel > 90 && sentinel < 95) || (sentinel > 122 && sentinel < 127) ||(sentinel == 96)){
+
+        /* Copy to clipboard */
+        guac_click_select_mark(click);
+
+        /* Draw selection */
+        click->select_head = click->select_col;
+        click->select_tail = click->select_col;
+
+        guac_click_draw_select(click);
+
+        return;
+
+    }
+
+    /* Blank */
+
+    else if (sentinel == 0 || sentinel == 32){
+
+        /* Get blank border */
+        guac_click* selection = guac_click_get_blank_border(click);
+
+        /* Blank position */
+        int head = selection->select_head;
+        int tail = selection->select_tail;
+        int width = selection->term->display->width;
+
+        if (head == 0 || tail < width - 1){
+            
+            /* get border */
+            guac_click* selection = guac_click_get_blank_border(click);
+
+            /* Copy to clipboard */
+            guac_click_select_blank(selection);
+
+            /* Draw selection */
+            guac_click_draw_select(selection);
+
+            return;
+        }
+
+        else if (head > 0 && tail == width - 1){
+
+            /* Move selected column */
+            sentinel = term->display->operations[row * term->display->width + col].character.value;
+            while(sentinel == 0 || sentinel == 32){
+                col--;
+                sentinel = term->display->operations[row * term->display->width + col].character.value;
+            }
+
+            selection->select_col = col;
+            selection->select_head = col;
+            selection->select_tail = col;
+            sentinel = term->display->operations[row * term->display->width + col].character.value;
+
+            /* Determination */
+
+            /* Words */
+
+            if ((sentinel > 64 && sentinel < 91) || (sentinel > 96 && sentinel < 123) || (sentinel > 47 && sentinel < 58) || (sentinel == 95)){
+                
+                /* get border */
+                guac_click* moved_selection = guac_click_get_word_border(selection);
+
+                /* Copy to clipboard */
+                guac_click_select_word(moved_selection);
+
+                /* Draw selection */
+                guac_click_draw_select(moved_selection);
+
+                return;
+
+            }
+
+            /* Marks */
+
+            else if ((sentinel > 32 && sentinel < 48) || (sentinel > 57 && sentinel < 65) || (sentinel > 90 && sentinel < 95) || (sentinel > 122 && sentinel < 127) ||(sentinel == 96)){
```

Comment:
There are a lot of magic numbers here and elsewhere in this function. Is there a better way to write this that would make things more readable and verifiable?